src/main/java/com/store/litflix/service/impl/CartServiceImpl.java

```diff
@@ -0,0 +1,116 @@
+package com.store.litflix.service.impl;
+
+import com.store.litflix.dto.cart.ShoppingCartRequestDto;
+import com.store.litflix.dto.cart.ShoppingCartResponseDto;
+import com.store.litflix.exception.CartItemAccessDeniedException;
+import com.store.litflix.exception.CartItemNotFoundException;
+import com.store.litflix.exception.EntityNotFoundException;
+import com.store.litflix.mapper.ShoppingCartMapper;
+import com.store.litflix.model.Book;
+import com.store.litflix.model.CartItem;
+import com.store.litflix.model.ShoppingCart;
+import com.store.litflix.model.User;
+import com.store.litflix.repository.book.BookRepository;
+import com.store.litflix.repository.cart.CartItemRepository;
+import com.store.litflix.repository.cart.CartRepository;
+import com.store.litflix.repository.user.UserRepository;
+import com.store.litflix.service.CartService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CartServiceImpl implements CartService {
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+    private final CartRepository cartRepository;
+    private final UserRepository userRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    @Override
+    public ShoppingCartResponseDto addBookToCart(ShoppingCartRequestDto requestDto) {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        User user = userRepository.findByEmail(authentication.getName())
+                .orElseThrow(() -> new RuntimeException("User not found"));
```

Comment:
provide parameter to message
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("User not found"));
```